cinema/views.py

```diff
@@ -2,0 +1,63 @@
+from rest_framework import viewsets
+
+from cinema.models import CinemaHall, Genre, Actor, Movie, MovieSession
+from cinema.serializers import (CinemaHallSerializer,
+                                GenreSerializer,
+                                ActorSerializer,
+                                MovieSerializer,
+                                MovieSessionSerializer,
+                                MovieListSerializer,
+                                MovieRetrieveSerializer,
+                                MovieSessionListSerializer,
+                                MovieSessionRetrieveSerializer)
+
+
+class CinemaHallViewSet(viewsets.ModelViewSet):
+    queryset = CinemaHall.objects.all()
+    serializer_class = CinemaHallSerializer
+
+
+class GenreViewSet(viewsets.ModelViewSet):
+    queryset = Genre.objects.all()
+    serializer_class = GenreSerializer
+
+
+class ActorViewSet(viewsets.ModelViewSet):
+    queryset = Actor.objects.all()
+    serializer_class = ActorSerializer
+
+
+class MovieViewSet(viewsets.ModelViewSet):
+    queryset = Movie.objects.all().prefetch_related("genres", "actors")
+    serializer_class = MovieSerializer
+
+    def get_serializer_class(self):
+        if self.action == "list":
+            return MovieListSerializer
+        elif self.action == "retrieve":
+            return MovieRetrieveSerializer
+        return MovieSerializer
```

Comment:
Consider adding type annotations to the `get_serializer_class` method to improve code readability and maintainability.